src/app/AppLayout/AppLayout.tsx

```diff
@@ -165,6 +167,42 @@ const AppLayout: React.FunctionComponent<IAppLayout> = ({children}) => {
   const handleAboutModalToggle = () => {
     setAboutModalOpen(!aboutModalOpen);
   };
+
+  React.useEffect(() => {
+    const sub = serviceContext.login.loggedIn().subscribe(
+      () => setShowUserIcon(true)
+    );
+    return () => sub.unsubscribe();
+  }, [serviceContext, serviceContext.login]);
+
+  React.useEffect(() => {
+    const sub = serviceContext.login.loggedOut().subscribe(
+      () => setShowUserIcon(false)
+    );
+    return () => sub.unsubscribe();
+  }, [serviceContext, serviceContext.login]);
+
+  const handleLogout = () => serviceContext.login.setLoggedOut();
+
+  const handleUserInfoToggle = () => setShowUserInfoDropdown(!showUserInfoDropdown);
+
+  React.useEffect(() => {
+    const sub = serviceContext.login.getUsername().subscribe(setUsername);
+    return () => sub.unsubscribe();
+  }, [serviceContext, serviceContext.login]);
+
+  const userInfoItems = [
+    <DropdownGroup key={0}>
+      <DropdownItem onClick={handleLogout}>Logout</DropdownItem>
+    </DropdownGroup>
+  ];
+
+  const UserInfoToggle = (
+    <DropdownToggle onToggle={handleUserInfoToggle} toggleIndicator={CaretDownIcon}>
+      {username ? username : <UserIcon color="white" size="sm"/>}
```

Comment:
Just a style thing, feel free to ignore it, but since the empty string is falsy, this could be shortened to `{ username || <UserIcon /> }`